nixos/modules/services/networking/networkmanager.nix

```diff
@@ -451,7 +451,7 @@ in {
 
     systemd.services.NetworkManager-dispatcher = {
       wantedBy = [ "network.target" ];
-      restartTriggers = [ configFile ];
+      restartTriggers = [ configFile overrideNameserversScript ];
```

Comment:
too bad we can't just watch on the directory, but this contains the nameserver list, so it should work.